core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1655,6 +1663,217 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleDescribeAcls(request: RequestChannel.Request): Unit = {
+    authorizeClusterAction(request)
+    val describeAclsRequest = request.body[DescribeAclsRequest]
+    authorizer match {
+      case None =>
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new DescribeAclsResponse(throttleTimeMs, new SecurityDisabledException(
+            "No Authorizer is configured on the broker."), Collections.emptySet[AclBinding]);
+        sendResponseMaybeThrottle(request, createResponse)
+      case Some(auth) =>
+        val filter = describeAclsRequest.filter()
+        var returnedAcls = new util.ArrayList[AclBinding]
+        val aclMap : Map[Resource, Set[Acl]] = auth.getAcls()
+        aclMap.foreach {
+          case (resource, acls) => {
+            acls.foreach {
+              case (acl) => {
+                val fixture = new AclBinding(new AdminResource(AdminResourceType.fromString(resource.resourceType.toString), resource.name),
+                    new AccessControlEntry(acl.principal.toString(), acl.host.toString(), acl.operation.toJava, acl.permissionType.toJava))
+                if (filter.matches(fixture))
+                  returnedAcls.add(fixture)
+              }
+            }
+          }
+        }
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new DescribeAclsResponse(throttleTimeMs, null, returnedAcls)
+        sendResponseMaybeThrottle(request, createResponse)
+    }
+  }
+
+  /**
+    * Convert an ACL binding filter to a Scala object.
+    * All ACL and resource fields must be specified (no UNKNOWN, ANY, or null fields are allowed.)
+    *
+    * @param filter     The binding filter as a Java object.
+    * @return           The binding filter as a scala object, or an exception if there was an error
+    *                   converting the Java object.
+    */
+  def toScala(filter: AclBindingFilter) : Try[(Resource, Acl)] = {
```

Comment:
Can we move some of this logic to a class in the `kafka.security.auth` package? Generally, we should do a better job of not cluttering `KafkaApis` too much since it needs to handle every protocol supported by the system.